### PR TITLE
WIP Method fromApi, fromAdmin, fromFront in request service

### PR DESCRIPTION
**Pull request by gillesbourgeat**
Hello,

This PR is a proposition for detect if the request comes from the api, admin or front
I have not found a method to do in Thelia.

The method name and content is not final.
All propositions are always welcome.

Cheers,

### CODE REVIEW

**Comment by lovenunu** [comment]
Well ... That's the same problem as find if the request is for the back office.
You can write the route ```/admin/foo``` and use a ```BaseFrontController```.
I don't think this is the solution.

**Comment by lunika** [comment]
this routes are not part of the API : 
```
    <route id="admin.configuration.api.downloadSecure" path="/admin/configuration/api/secure_key/{api_id}">
        <default key="_controller">Thelia\Controller\Admin\ApiController::downloadAction</default>
    </route>

    <route id="admin.configuration.api.delete" path="/admin/configuration/api/delete">
        <default key="_controller">Thelia\Controller\Admin\ApiController::deleteAction</default>
    </route>

    <route id="admin.configuration.api.update" path="/admin/configuration/api/update/{api_id}" methods="get">
        <default key="_controller">Thelia\Controller\Admin\ApiController::updateAction</default>
    </route>

    <route id="admin.configuration.api.update.process" path="/admin/configuration/api/update/{api_id}" methods="post">
        <default key="_controller">Thelia\Controller\Admin\ApiController::processUpdateAction</default>
    </route>
```

But your method will return true.

**Comment by lovenunu** [comment]
s/>=/===/g

File: core/lib/Thelia/Core/HttpFoundation/Request.php [hunk 1]
```diff
@@ -87,4 +94,51 @@ public function toString($withContent = true)
...
+    }
+
+    /**
+     * Detect if the request comes from the api
+     *
+     * @return bool
+     */
+    public function isApi()
+    {
+        if ($this->controllerType === self::CONTROLLER_API) {
```
Comment (lunika) [inline core/lib/Thelia/Core/HttpFoundation/Request.php]: you can directly put : 

```
return $this->controllerType === self::CONTROLLER_API;
```

It's the exact same thing.

**Comment by gillesbourgeat** [comment]
@lunika indeed, rebase ok.

File: core/lib/Thelia/Core/HttpFoundation/Request.php (hunk 1 above)
Comment (gillesbourgeat) [inline core/lib/Thelia/Core/HttpFoundation/Request.php]: @lunika indeed, rebase ok.

**Comment by lunika** [comment]
:+1:

**Pull request merged**